dataset/test/bin_test.cpp

```diff
@@ -206,3 +208,15 @@ TEST_P(BinTest, group_and_bin) {
   const auto group = bin(table, {}, {groups});
   EXPECT_EQ(bin(group, {edges_x}, {}), x_group);
 }
+
+TEST_P(BinTest, rebin_masked) {
+  const auto table = GetParam();
+  auto binned = bin(table, {edges_x_coarse});
+  binned.masks().set("x-mask", makeVariable<bool>(Dims{Dim::X}, Shape{2},
+                                                  Values{false, true}));
+  if (table.dims().volume() > 0) {
+    EXPECT_NE(bin(binned, {edges_x}), bin(table, {edges_x}));
```

Comment:
Might be instructive for future maintenance to demonstrate the effect of dropping the mask and reapplying the comparison.